### PR TITLE
[FIX] payment: update modules that depend on `account`

### PR DESCRIPTION
**Pull request by adwid**
On some DB, it is not possible to update some modules that depend on
`account`

To reproduce the issue:
1. Checkout before [1] and install `account_accountant`
2. Checkout after [1]
3. Update `account_accountant`

Error: a server error is displayed

> Field 'code' used in domain of field 'payment_acquirer_id' ([
> ('provider', '=', code)]) must be present in view but is missing."")

Commit [1] contains unstable changes: it adds a domain on the field
`payment_acquirer_id` of `account.payment.method.line` and, to make
it work, it also adds the needed field `code` on the view
`view_account_journal_form`. But, if the view is not updated on an
existing database, the domain can not work, hence the client error.

This explains why the above use case does not work:
The module `account_accountant` has a view that also inherits the
view of account `account.view_account_journal_form`. As a result,
when updating the module (and, therefore, the view of
`account_accountant`), we `:View._validate_view` the whole view, which
will lead to:
https://github.com/odoo/odoo/blob/2cbf9d840b79520854731fd6a23431c69e451775/odoo/addons/base/models/ir_ui_view.py#L2282-L2286

[1] https://github.com/odoo-dev/odoo/commit/88782d3a76fb3941b9ed7d998ec0610aa26ff611

opw-3993709
opw-3993707
sentry-5416409171

### CODE REVIEW

**Comment by adwid** [comment]
The fix is supposed to be an imitation of the domain, but I'm even 
wondering whether the domain is correct or not. On a new DB with 
`account`, open the payment acquirer "Wire Transfer", it leads to 
the compute of its `journal_id`. In this method, we trigger an ensure:
https://github.com/odoo/odoo/blob/88782d3a76fb3941b9ed7d998ec0610aa26ff611/addons/payment/models/payment_acquirer.py#L224-L225
And it will lead to the creation of a payment method line:
https://github.com/odoo/odoo/blob/88782d3a76fb3941b9ed7d998ec0610aa26ff611/addons/payment/models/payment_acquirer.py#L198-L204
But... With the method _Manual_ (code: `manual`) and the current 
acquirer, i.e. _'Wire Transfer'_ (provider: `transfer`). As here we 
are not talking about a constraint, I guess we could consider it as 
ok, but still, I think it's worth a mention and let the reviewers 
decide what to do next.

**Comment by adwid** [comment]
robodoo up to saas-17.2
17.3 did not exist when 88782d3a76fb3941b9ed7d998ec0610aa26ff611 has been fw-ed on master (see https://github.com/odoo/odoo/pull/166745#event-12947183701)

**Review by agr-odoo — COMMENTED** [review]
tl;dr
Current patch looks good but a simpler fix could be to just move the domain to the view

-----

I'm checking the PR but I can't reproduce with the provided steps, in particular the step
```
In Incoming Payments, enable the column "Payment Acquirer"
```
Seems wrong, as we don't have the column "Payment Acquirer" without updating the view, after which we should not have issues anymore

I could reproduce the issue by upgrading another module ( a localization module or `account_accountant`)
```
Error while validating view near:

<form string="Account Journal" __validate__="1">
                    <sheet>
                        <div name="button_box" class="oe_button_box">

Field 'code' used in domain of field 'payment_acquirer_id' ([('provider', '=', code)]) must be present in view but is missing.
```
And indeed it seems to occur because the system attempt to validate the (not yet updated) base view, where `payment_acquirer_id` python domain is referencing a field `code` not existing in the view.
https://github.com/odoo/odoo/blob/6afd72b6a377d57ce46b643544931b0643e5d7a9/odoo/addons/base/models/ir_ui_view.py#L1374-L1386
This occurs because the domain has been set in the python code and it is accessible even before we have `code` in the view, so imho a possible solution is to move the domain to the view

**Comment by adwid** [comment]
@agr-odoo 
Yeah my bad, I did a small error with the steps, I have updated everything
(And considering the new situation, indeed, we can simply move the domain on the view)

**Review by agr-odoo — APPROVED** [review]
Looks good to me :+1:

**Comment by agr-odoo** [comment]
@smetl 
Hello,
Could you give a quick look too?

**Review by smetl — APPROVED** [review]
@robodoo r+
